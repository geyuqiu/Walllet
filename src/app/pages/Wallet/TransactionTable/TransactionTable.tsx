import React from "react";

import {Table} from '../../../components/Table/Table';
import {Wallet} from '../model';
import {Transaction} from '../TransactionRow/model';
import {feeAccessor, tableColumns, timestampAccessor} from '../TransactionRow/TableColumns';
import {TransactionRow} from '../TransactionRow/TransactionRow';

export interface HideColumnsResponsive {
	accessor: string;
	breakPoint: number;
}

const hideColumnsAtBreakpoint: HideColumnsResponsive[] = [
	{accessor: timestampAccessor, breakPoint: 1280},
	{accessor: feeAccessor, breakPoint: 1024}
];

type AddressDropdownProps = {
	wallet: Wallet | null;
	transactions: Transaction[];
	isLoadingTransactions: boolean;
}

export const TransactionTable = ({wallet, transactions, isLoadingTransactions}: AddressDropdownProps) => (
	<div className="ml-6 mr-3 sm:mx-8 sm:flex sm:justify-center pt-12">
		{(isLoadingTransactions || !wallet) && <p data-testid="TransactionTable__Loading">Loading Transactions ...</p>}
		{!isLoadingTransactions && !transactions?.length && wallet && <p data-testid="TransactionTable__No_Transactions">No transactions were found for this wallet!</p>}
		{!isLoadingTransactions && transactions?.length > 0 && wallet &&
		<Table columns={tableColumns} data={transactions} hideColumnsAtBreakpoint={hideColumnsAtBreakpoint}>
			{(transaction: Transaction) => (
				<TransactionRow transaction={transaction} address={wallet.address}/>
			)}
		</Table>
		}
	</div>
);
