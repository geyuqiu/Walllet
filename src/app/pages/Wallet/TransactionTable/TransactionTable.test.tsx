import {render} from "@testing-library/react";
import React from "react";

import {TransactionTable} from "./TransactionTable";

describe("TransactionTable", () => {
	it("isLoadingTransactions should match snapshot", () => {
		const {getAllByTestId, container} = render(<TransactionTable isLoadingTransactions/>);
		expect(getAllByTestId("TransactionTable__Loading")).toHaveLength(1);
		expect(container).toMatchSnapshot();
	});

	it("transactions empty for selected address should match snapshot", () => {
		const {getAllByTestId, container} = render(<TransactionTable
			transactions={[]} isLoadingTransactions={false}
			wallet={{
				address: 'AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW',
				balance: '0'
			}}/>);

		expect(getAllByTestId("TransactionTable__No_Transactions")).toHaveLength(1);
		expect(container).toMatchSnapshot();
	});

	it("display n transactions should match snapshot", () => {
		const transactions = [
			{
				id: "6eba34e4513918928313567d5c60fcb9635f75dbb0af0164e58c59216efe6749",
				blockId: "8b03f113fc3ee743411a3be687db96ab312e0e6be10e9a71eb9c705e04be4968",
				version: 2,
				type: 0,
				typeGroup: 1,
				amount: "1004310900000",
				fee: "1000000",
				sender: "AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv",
				senderPublicKey: "03f46c15d483e099b8c1ad277aa0341fd24e4052a835f848d72ca8632d1f16cf60",
				recipient: "AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW",
				signature: "095e77ea3c8072a80f8855d114abbf4a66cfa700918b50305390eafd19948df7d6edfb2ede299d857f4bf95daceecc990729b0194d80575f2cf92bf61e1d0895",
				confirmations: 17615,
				timestamp: "10.04.2021",
				nonce: 1544
			},
			{
				id: "b4f53e85e7a6ec275663bfbfc4e4f5a1394a2d6f9b36626b63488f39426bbe19",
				blockId: "dd3e223903d78436d16b2f95a3d47f9f6732718770625f84ee5a7c1d9e9e3202",
				version: 2,
				type: 0,
				typeGroup: 1,
				amount: "1046145300000",
				fee: "1000000",
				sender: "AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv",
				senderPublicKey: "03f46c15d483e099b8c1ad277aa0341fd24e4052a835f848d72ca8632d1f16cf60",
				recipient: "AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW",
				signature: "079908f962274da5a2b01b6c36cb81b3367542be1733d93f9e405ed98ea9b0f8adb22f7b1d8233f2837fd740a7933d354a74caa4a4fe79b7f1a522965de0da79",
				confirmations: 17616,
				timestamp: "10.04.2021",
				nonce: 1543
			},
			{
				id: "3f3f28f3d96b19e3a8424b7d0a1fa13a00d2bd1dca8a564a587c39ea9ec5cccc",
				blockId: "db694d8d64cce9a4de5902725a56ec4314842596acd2b2becb0f45b08e1288ff",
				version: 2,
				type: 0,
				typeGroup: 1,
				amount: "1047909000000",
				fee: "1000000",
				sender: "AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv",
				senderPublicKey: "03f46c15d483e099b8c1ad277aa0341fd24e4052a835f848d72ca8632d1f16cf60",
				recipient: "AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW",
				signature: "9be24d4cec81c2f6efcdba2b0bd452d97ca35bc9c80f0fb398998e11b1396bf4055af6b58aec9e5fbc6de38d2f7bf4b4f2b73766ffbd37425b03718ccf23ad49",
				confirmations: 17617,
				timestamp: "10.04.2021",
				nonce: 1542
			},
			{
				id: "a446aa06a4f3986c3b042d88c066d798d76a40af9478826f22762e5f91122270",
				blockId: "6454cebac39ba3785df2f58491ca4984451772a2cf26384ce7e7a8acf657b5da",
				version: 2,
				type: 0,
				typeGroup: 1,
				amount: "1054332400000",
				fee: "1000000",
				sender: "AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv",
				senderPublicKey: "03f46c15d483e099b8c1ad277aa0341fd24e4052a835f848d72ca8632d1f16cf60",
				recipient: "AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW",
				signature: "25f61c80fada6c1eeed6c437aa09cafddeaa92dbc0465a5bc4274b63bc282b5ffb91c1ba7e55542fb3677d772a4eb684c55e9b74cfe419fee4d16f4e56ad4e84",
				confirmations: 17618,
				timestamp: "10.04.2021",
				nonce: 1541
			}
		];
		const {getAllByTestId, container} = render(<TransactionTable
			transactions={transactions} isLoadingTransactions={false}
			wallet={{
				address: 'AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW',
				balance: '0'
			}}/>);

		expect(getAllByTestId("TableRow")).toHaveLength(4);
		expect(container).toMatchSnapshot();
	});

});
