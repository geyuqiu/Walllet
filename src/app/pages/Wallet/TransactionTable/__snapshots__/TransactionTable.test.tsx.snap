// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionTable display n transactions should match snapshot 1`] = `
<div>
  <div
    class="ml-6 mr-3 sm:mx-8 sm:flex sm:justify-center pt-12"
  >
    <div
      class="Table__TableWrapper-ck3yk8-0 kEIFJE w-full max-w-screen-xl -mt-3"
      role="table"
    >
      <table
        cellpadding="0"
        class="table-auto"
      >
        <thead>
          <tr
            class="border border-theme-secondary-300"
            role="row"
          >
            <th
              class="group border text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 m-0 p-3 md:px-5 font-semibold hasBorder"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top"
              >
                <div
                  data-testid="table__th--0"
                >
                  Txid
                </div>
                <div
                  class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
                >
                  <div
                    class="sc-bdfBwQ kJrqvM transition-transform transform rotate-180"
                    height="5"
                    role="img"
                    width="8"
                  >
                    <svg>
                      arrow-down.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="group border text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 m-0 p-3 md:px-5 font-semibold hasBorder"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top"
              >
                <div
                  data-testid="table__th--1"
                >
                  Sender
                </div>
                <div
                  class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
                >
                  <div
                    class="sc-bdfBwQ kJrqvM transition-transform transform rotate-180"
                    height="5"
                    role="img"
                    width="8"
                  >
                    <svg>
                      arrow-down.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="group border text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 m-0 p-3 md:px-5 font-semibold hasBorder"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top"
              >
                <div
                  data-testid="table__th--2"
                >
                  Recipient
                </div>
                <div
                  class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
                >
                  <div
                    class="sc-bdfBwQ kJrqvM transition-transform transform rotate-180"
                    height="5"
                    role="img"
                    width="8"
                  >
                    <svg>
                      arrow-down.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="group border text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 m-0 p-3 md:px-5 font-semibold hasBorder"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top justify-end flex-row-reverse"
              >
                <div
                  data-testid="table__th--3"
                >
                  Amount
                </div>
                <div
                  class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-auto mr-2"
                >
                  <div
                    class="sc-bdfBwQ kJrqvM transition-transform transform rotate-180"
                    height="5"
                    role="img"
                    width="8"
                  >
                    <svg>
                      arrow-down.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="group border text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 m-0 p-3 md:px-5 font-semibold hasBorder"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top justify-end flex-row-reverse"
              >
                <div
                  data-testid="table__th--4"
                >
                  Fee
                </div>
                <div
                  class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-auto mr-2"
                >
                  <div
                    class="sc-bdfBwQ kJrqvM transition-transform transform rotate-180"
                    height="5"
                    role="img"
                    width="8"
                  >
                    <svg>
                      arrow-down.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="TableRow__TableRowStyled-sc-1p89o19-0 iRdKeT"
            data-testid="TableRow"
          >
            <td
              class="md:py-2 md:w-4"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start"
              >
                <a
                  class="text-green-dark hover:text-green-darkest active:text-green-light font-semibold hover:underline"
                  href="https://explorer.ark.io/transaction/6eba34e4513918928313567d5c60fcb9635f75dbb0af0164e58c59216efe6749"
                >
                  <span
                    class="table-cell lg:table-cell md:hidden"
                  >
                    6eba3...e6749
                  </span>
                  <div
                    class="sc-bdfBwQ iyevwF hidden lg:hidden md:table-cell"
                    height="20"
                    width="20"
                  >
                    <svg>
                      txid.svg
                    </svg>
                  </div>
                </a>
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start md:border-l border-theme-secondary-300"
              >
                <a
                  class="text-green-dark hover:text-green-darkest active:text-green-light font-semibold hover:underline"
                  href="https://explorer.ark.io/wallets/AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv"
                >
                  <span
                    class="hidden sm:block md:hidden"
                  >
                    AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv
                  </span>
                  <span
                    class="block sm:hidden md:block"
                  >
                    AVyqD...LhRjv
                  </span>
                </a>
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start md:border-l border-theme-secondary-300"
              >
                <span
                  class="hidden sm:block md:hidden"
                >
                  AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW
                </span>
                <span
                  class="block sm:hidden md:block"
                >
                  Adzbh...FgETuW
                </span>
              </div>
            </td>
            <td
              class="hidden xl:table-cell"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-center md:border-l border-theme-secondary-300"
              >
                10.04.2021
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:border-l border-theme-secondary-300"
              >
                <span
                  data-testid="transaction__fee__Amount"
                >
                  10,043.109 DARK
                </span>
              </div>
            </td>
            <td
              class="hidden lg:table-cell"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:border-l border-theme-secondary-300"
              >
                <span
                  data-testid="transaction__fee__Amount"
                >
                  0.01 DARK
                </span>
              </div>
            </td>
          </tr>
          <tr
            class="TableRow__TableRowStyled-sc-1p89o19-0 iRdKeT"
            data-testid="TableRow"
          >
            <td
              class="md:py-2 md:w-4"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start"
              >
                <a
                  class="text-green-dark hover:text-green-darkest active:text-green-light font-semibold hover:underline"
                  href="https://explorer.ark.io/transaction/b4f53e85e7a6ec275663bfbfc4e4f5a1394a2d6f9b36626b63488f39426bbe19"
                >
                  <span
                    class="table-cell lg:table-cell md:hidden"
                  >
                    b4f53...bbe19
                  </span>
                  <div
                    class="sc-bdfBwQ iyevwF hidden lg:hidden md:table-cell"
                    height="20"
                    width="20"
                  >
                    <svg>
                      txid.svg
                    </svg>
                  </div>
                </a>
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start md:border-l border-theme-secondary-300"
              >
                <a
                  class="text-green-dark hover:text-green-darkest active:text-green-light font-semibold hover:underline"
                  href="https://explorer.ark.io/wallets/AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv"
                >
                  <span
                    class="hidden sm:block md:hidden"
                  >
                    AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv
                  </span>
                  <span
                    class="block sm:hidden md:block"
                  >
                    AVyqD...LhRjv
                  </span>
                </a>
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start md:border-l border-theme-secondary-300"
              >
                <span
                  class="hidden sm:block md:hidden"
                >
                  AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW
                </span>
                <span
                  class="block sm:hidden md:block"
                >
                  Adzbh...FgETuW
                </span>
              </div>
            </td>
            <td
              class="hidden xl:table-cell"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-center md:border-l border-theme-secondary-300"
              >
                10.04.2021
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:border-l border-theme-secondary-300"
              >
                <span
                  data-testid="transaction__fee__Amount"
                >
                  10,461.453 DARK
                </span>
              </div>
            </td>
            <td
              class="hidden lg:table-cell"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:border-l border-theme-secondary-300"
              >
                <span
                  data-testid="transaction__fee__Amount"
                >
                  0.01 DARK
                </span>
              </div>
            </td>
          </tr>
          <tr
            class="TableRow__TableRowStyled-sc-1p89o19-0 iRdKeT"
            data-testid="TableRow"
          >
            <td
              class="md:py-2 md:w-4"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start"
              >
                <a
                  class="text-green-dark hover:text-green-darkest active:text-green-light font-semibold hover:underline"
                  href="https://explorer.ark.io/transaction/3f3f28f3d96b19e3a8424b7d0a1fa13a00d2bd1dca8a564a587c39ea9ec5cccc"
                >
                  <span
                    class="table-cell lg:table-cell md:hidden"
                  >
                    3f3f2...5cccc
                  </span>
                  <div
                    class="sc-bdfBwQ iyevwF hidden lg:hidden md:table-cell"
                    height="20"
                    width="20"
                  >
                    <svg>
                      txid.svg
                    </svg>
                  </div>
                </a>
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start md:border-l border-theme-secondary-300"
              >
                <a
                  class="text-green-dark hover:text-green-darkest active:text-green-light font-semibold hover:underline"
                  href="https://explorer.ark.io/wallets/AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv"
                >
                  <span
                    class="hidden sm:block md:hidden"
                  >
                    AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv
                  </span>
                  <span
                    class="block sm:hidden md:block"
                  >
                    AVyqD...LhRjv
                  </span>
                </a>
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start md:border-l border-theme-secondary-300"
              >
                <span
                  class="hidden sm:block md:hidden"
                >
                  AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW
                </span>
                <span
                  class="block sm:hidden md:block"
                >
                  Adzbh...FgETuW
                </span>
              </div>
            </td>
            <td
              class="hidden xl:table-cell"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-center md:border-l border-theme-secondary-300"
              >
                10.04.2021
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:border-l border-theme-secondary-300"
              >
                <span
                  data-testid="transaction__fee__Amount"
                >
                  10,479.09 DARK
                </span>
              </div>
            </td>
            <td
              class="hidden lg:table-cell"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:border-l border-theme-secondary-300"
              >
                <span
                  data-testid="transaction__fee__Amount"
                >
                  0.01 DARK
                </span>
              </div>
            </td>
          </tr>
          <tr
            class="TableRow__TableRowStyled-sc-1p89o19-0 iRdKeT"
            data-testid="TableRow"
          >
            <td
              class="md:py-2 md:w-4"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start"
              >
                <a
                  class="text-green-dark hover:text-green-darkest active:text-green-light font-semibold hover:underline"
                  href="https://explorer.ark.io/transaction/a446aa06a4f3986c3b042d88c066d798d76a40af9478826f22762e5f91122270"
                >
                  <span
                    class="table-cell lg:table-cell md:hidden"
                  >
                    a446a...22270
                  </span>
                  <div
                    class="sc-bdfBwQ iyevwF hidden lg:hidden md:table-cell"
                    height="20"
                    width="20"
                  >
                    <svg>
                      txid.svg
                    </svg>
                  </div>
                </a>
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start md:border-l border-theme-secondary-300"
              >
                <a
                  class="text-green-dark hover:text-green-darkest active:text-green-light font-semibold hover:underline"
                  href="https://explorer.ark.io/wallets/AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv"
                >
                  <span
                    class="hidden sm:block md:hidden"
                  >
                    AVyqDoLngpCnmqDtEicmYcNbezAKPLhRjv
                  </span>
                  <span
                    class="block sm:hidden md:block"
                  >
                    AVyqD...LhRjv
                  </span>
                </a>
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-start md:border-l border-theme-secondary-300"
              >
                <span
                  class="hidden sm:block md:hidden"
                >
                  AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW
                </span>
                <span
                  class="block sm:hidden md:block"
                >
                  Adzbh...FgETuW
                </span>
              </div>
            </td>
            <td
              class="hidden xl:table-cell"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:justify-center md:border-l border-theme-secondary-300"
              >
                10.04.2021
              </div>
            </td>
            <td>
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:border-l border-theme-secondary-300"
              >
                <span
                  data-testid="transaction__fee__Amount"
                >
                  10,543.324 DARK
                </span>
              </div>
            </td>
            <td
              class="hidden lg:table-cell"
            >
              <div
                class="TableCell__TableCellInnerWrapper-sc-12y9mmp-0 hQDHYv justify-end md:border-l border-theme-secondary-300"
              >
                <span
                  data-testid="transaction__fee__Amount"
                >
                  0.01 DARK
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`TransactionTable isLoadingTransactions should match snapshot 1`] = `
<div>
  <div
    class="ml-6 mr-3 sm:mx-8 sm:flex sm:justify-center pt-12"
  >
    <p
      data-testid="TransactionTable__Loading"
    >
      Loading Transactions ...
    </p>
  </div>
</div>
`;

exports[`TransactionTable transactions empty for selected address should match snapshot 1`] = `
<div>
  <div
    class="ml-6 mr-3 sm:mx-8 sm:flex sm:justify-center pt-12"
  >
    <p
      data-testid="TransactionTable__No_Transactions"
    >
      No transactions were found for this wallet!
    </p>
  </div>
</div>
`;
